TombEngine/Specific/EngineMain.cpp

```diff
@@ -0,0 +1,521 @@
+{
+	std::vector<Vector2i> resList;
+
+	SDL_DisplayID display = SDL_GetPrimaryDisplay();
+	if (!display)
+		return resList;
+
+	int count = 0;
+	SDL_DisplayMode** modes = SDL_GetFullscreenDisplayModes(display, &count);
+	if (!modes || count <= 0)
+		return resList;
+
+	resList.reserve(count);
+
+	for (int i = 0; i < count; ++i)
+	{
+		const SDL_DisplayMode* mode = modes[i];
+		if (!mode)
+			continue;
+
+		Vector2i res{ mode->w, mode->h };
+
+		bool add = true;
+		for (const auto& m : resList)
+		{
+			if (m.x == res.x && m.y == res.y)
+			{
+				add = false;
+				break;
+			}
+		}
+
+		if (add)
+			resList.push_back(res);
+	}
+
+	SDL_free(modes);
+
+	std::sort(
+		resList.begin(), resList.end(),
+		[](const Vector2i& a, const Vector2i& b)
+		{
+			return (a.x == b.x) ? (a.y < b.y) : (a.x < b.x);
+		});
+
+	return resList;
+}
+
+int SDLCALL ConsoleInput(void*)
+{
+	auto input = std::string();
+	while (!ThreadEnded)
+	{
+		if (!std::getline(std::cin, input))
+			break;
+
+		if (std::regex_match(input, std::regex("^\\s*$")))
+			continue;
+
+		if (g_GameScript == nullptr)
+		{
+			TENLog("Scripting engine not initialized.", LogLevel::Error);
+			continue;
+		}
+		else
+		{
+			g_GameScript->AddConsoleInput(input);
+		}
+	}
+
+	return 0;
+}
+
+static void HandleWindowFocusGained(SDL_Window* window)
+{
+	SetInputLockState(false);
+
+	if (!g_Configuration.EnableWindowedMode)
+	{
+		g_Renderer.ToggleFullScreen(true);
+	}
+
+	if (ThreadSuspendCount > 0)
+	{
+		TENLog("Resuming game thread", LogLevel::Info);
+
+		if (!g_VideoPlayer.Resume())
+			ResumeAllSounds(SoundPauseMode::Global);
+
+		ResumeGameThread();
+	}
+}
+
+static void HandleWindowFocusLost(SDL_Window* window)
+{
+	SetInputLockState(true);
+
+	if (!g_Configuration.EnableWindowedMode)
+		SDL_MinimizeWindow(window);
+
+	bool isMinimized =
+		(SDL_GetWindowFlags(window) & SDL_WINDOW_MINIMIZED) != 0;
+
+	if ((!DebugMode || isMinimized) && ThreadSuspendCount == 0)
+	{
+		TENLog("Suspending game thread", LogLevel::Info);
+
+		if (!g_VideoPlayer.Pause())
+			PauseAllSounds(SoundPauseMode::Global);
+
+		PauseGameThread();
+	}
+}
+
+void PauseGameThread()
+{
+	SDL_LockMutex(GamePauseMutex);
+
+	ThreadSuspendCount++;
+	GamePaused = true;
+
+	SDL_UnlockMutex(GamePauseMutex);
+}
+
+void ResumeGameThread()
+{
+	SDL_LockMutex(GamePauseMutex);
+
+	if (ThreadSuspendCount > 0)
+		ThreadSuspendCount--;
+
+	if (ThreadSuspendCount == 0)
+	{
+		GamePaused = false;
+		SDL_BroadcastCondition(GamePauseCond);
+	}
+
+	SDL_UnlockMutex(GamePauseMutex);
+}
+
+void WaitIfGamePaused()
+{
+	SDL_LockMutex(GamePauseMutex);
+	while (GamePaused && !ThreadEnded && DoTheGame)
+	{
+		SDL_WaitCondition(GamePauseCond, GamePauseMutex);
+	}
+	SDL_UnlockMutex(GamePauseMutex);
+}
+
+int main(int argc, char* argv[])
+{
+	g_Platform = CreatePlatformSubsystem();
+	g_Platform->Initialize();
+	g_Platform->CheckPrerequisites();
+	
+	// Initialize SDL3
+	if (!SDL_Init(SDL_INIT_VIDEO | SDL_INIT_AUDIO | SDL_INIT_EVENTS))
```

Comment:
Why are we initializing audio and video SDL subsystems, if they aren't used? Or video subsystem is already used?